### ISSUE TITLE
Accessibility

### ISSUE THREAD

**Issue by majkoll** [cyntler/hamburger-react#74]
The toggle button provided is currently rendered as a div element instead of a button element. This approach has accessibility implications as it lacks the necessary ARIA attributes and role needed for proper accessibility support.

For accessibility, it would be better if the toggle button is rendered as a button element. If it must be a div, appropriate ARIA attributes and roles should be added to ensure it is accessible (e.g. role, aria-pressed).

**Comment by ARXChrono** [comment]
<img width="1007" alt="Image" src="https://github.com/user-attachments/assets/0f5d6275-6629-4079-a378-90b5bfa913f8" />

https://pagespeed.web.dev/analysis/https-hamburger-react-netlify-app/b94bw07dtv?form_factor=mobile
